waku/v2/node/wakunode2.nim

```diff
@@ -800,7 +800,9 @@ proc start*(node: WakuNode) {.async.} =
   # TODO Get this from WakuNode obj
   let peerInfo = node.peerInfo
   info "PeerInfo", peerId = peerInfo.peerId, addrs = peerInfo.addrs
-  let listenStr = $peerInfo.addrs[^1] & "/p2p/" & $peerInfo.peerId
+  let listenStr = if websocketConn: @[$peerInfo.addrs[^1] & "/p2p/" & $peerInfo.peerId,
```

Comment:
This seems a bit error-prone, especially if we have more transports. What else is in addrs field? Can we iterate over this and add them all?

Might be worth to see what Nimbus or other libp2p consumers do here

Reply:
We have host address along with announced address.
Iterating over them may not be the best idea as we have to add peerID separately anyway, however it might be a good idea to make the index constants .

Reply:
We have this outstanding issue to remove reliance on our own cached `peerInfo`: https://github.com/status-im/nim-waku/issues/744

We don't have to remove `peerInfo` in this PR, but I think a better intermediate step would be to cache the `announcedAddresses` on the node to keep them separate from the host addresses. This way you can just iterate over the announcedAddresses for the listenStr (`info` would have to be updated as well), without requiring the extra state variable in for the node.

Reply:
resolved in 7c7f9fb106ac162e7efef4a3ae29daa5bc9110fe